### DISCUSSION TITLE
Defaults, better h26X format support and hw accelerated encoding

### DISCUSSION THREAD

**Discussion by danboid** [SeaDve/Kooha#181] [Ideas]
Hi @SeaDve 

Thanks for KooHa! I have been waiting for a simple to use, Wayland compatible screen capture tool that actually works for a decade or so and Kooha is that app so it seems I can finally attempt to fully switch from X to a Wayland based desktop (most likely GNOME + extensions like dock2panel etc) and potentially not have any notable missing features that I make regular use of under X11 such as easy screen capture via simplescreenrecorder.

Kooha defaults

The main reason I have started this discussion is regarding the current Kooha defaults, primarily the default state of the sound recording options. I think Kooha should default to having the recording of desktop sounds enabled by default, but not the microphone. I made the mistake of coming onto GH to file an issue, thinking it was a snap problem, before realising I was getting no audio on my test captures because I'd not enabled desktop audio capture. I think its pretty safe to assume most users also want to record audio. I'd argue its a better default situation to have the option of stripping the audio from a capture if you didn't want it rather than never being able to record it again in the case of capturing one-off live streams. For these reasons I'd like to see desktop audio recording enabled by default.

h264 and h265 support

I've not got Kooha installed on the machine I'm on atm so I can't find out if the kooha mkv format option encodes to but I would presume expect it is using either h264 or h265, or something else. Whatever it does I'd like to see Kooha offer the option of using either h26X codec. Kooha needs to be able to record to both h264 and h265 IMO as these are still the most well supported consumer codecs. There are reasons why users may prefer one format over the other still and everyone wants to avoid having to re-encode video if possible.

HW accelerated encoding

Another potential feature request - does Kooha currently show you when its using hw accelerated encoding? Which GPUs does kooha support hw accelerated video encoding on? Does this work under both Wayland and X?

**Comment by SeaDve** [comment]
> Another potential feature request - does Kooha currently show you when its using hw accelerated encoding?

Currently none, through the UI. But it should be printed on the terminal what encoder was used.

> Which GPUs does Kooha support HW accelerated video encoding on?

Kooha currently doesn't support HW accelerated video encoding, that's why it's hidden behind environment variables, however, HW accelerated should work on most Intel and AMD GPUs

> Does this work under both Wayland and X?

If it works on Wayland, it will likely work on X.

Regarding codecs and containers, I've been working on a profile editor window in the `encodebin` branch (very WIP, and might change the UI). This would allow the user to choose any encoder available in their system.

![image](https://user-images.githubusercontent.com/64297935/187140821-925d31bc-370b-41dd-af10-dce5313ff4b5.png)

**Comment by SeaDve** [comment]
Kooha now defaults with desktop audio enabled in `2.2.0`
